benchmark/utils/cuda_linops.hpp

```diff
@@ -370,14 +355,13 @@ class CuspCsrEx
             csr_->get_num_stored_elements(), &alpha, this->get_descr(),
             csr_->get_const_values(), csr_->get_const_row_ptrs(),
             csr_->get_const_col_idxs(), db, &beta, dx, &buffer_size);
-        GKO_ASSERT_NO_CUDA_ERRORS(cudaMalloc(&buffer_, buffer_size));
-        set_buffer_ = true;
+        buffer_.resize_and_reset(buffer_size);
```

Comment:
If buffersize does not depend on b/x data, putting it in the read stage may be reasonable.